src/components/RichTextEditor/index.tsx

```diff
@@ -1,24 +1,67 @@
-import React from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 import { useEditor, EditorContent } from '@tiptap/react';
 import StarterKit from '@tiptap/starter-kit';
 import Image from '@tiptap/extension-image';
 import Table from '@tiptap/extension-table';
 import TableRow from '@tiptap/extension-table-row';
 import TableCell from '@tiptap/extension-table-cell';
 import TableHeader from '@tiptap/extension-table-header';
+import Underline from '@tiptap/extension-underline';
+import Placeholder from '@tiptap/extension-placeholder';
 import styles from './RichTextEditor.css';
 import classNames from 'classnames';
 import MenuBar from './MenuBar';
 import { MENU_EDIT_OPTIONS } from './richTextEditOptions';
 
 interface Props {
-  menuEditOptions?: typeof MENU_EDIT_OPTIONS;
+  displayOnlyOnFocus?: boolean;
   editorContent?: string;
+  menuEditOptions?: typeof MENU_EDIT_OPTIONS;
+  menuPositionedBottom?: boolean;
+  placeholderText?: string;
   variablesClassName?: string;
 }
 
 const RichTextEditor: React.FC<Props> = props => {
-  const { menuEditOptions, variablesClassName, editorContent } = props;
+  const {
+    displayOnlyOnFocus,
+    editorContent,
+    menuEditOptions,
+    menuPositionedBottom,
+    placeholderText,
+    variablesClassName
+  } = props;
+
+  const [renderMenuBar, setRenderMenuBar] = displayOnlyOnFocus ? useState(false) : useState(true);
+
+  const buttonsMenuElement = useRef<HTMLDivElement>(null);
+  const textAreaElement = useRef<HTMLDivElement>(null);
+
+  const buttonsMenuConcealer = () => {
```

Comment:
Can you please check if it is correct to have a hook inside a function? It is the first time I see it and it feels weird

Reply:
It's not, just checked here, thanks for the correction.
I updated it.